### ISSUE TITLE
duplicate header "Content-Length" in HTTP responses

### ISSUE THREAD

**Issue by sqlnew** [jchristn/HttpServerLite#7]
In the Content Routes, When a folder route is specified and the file address is accessed, the HTTP header will return a duplicate "Content-Length" in HTTP responses.
When Test in the postman, it will cause errors: Parse Error: "The response has a duplicate "Content-Length" header"
If the author confirms that this is a bug, please fix it. Thank you very much.

**Comment by jchristn** [comment]
Hi @sqlnew could you help me reproduce?  I just tried this using the ```Test``` project and only see a single instance of ```Content-Length``` when using content routes.

![image](https://user-images.githubusercontent.com/1421081/146570532-bb5e1582-ca41-424a-b1bd-65c81535ec06.png)

cURL:
```
C:\Users\joelc>curl -v http://localhost:8080/html/index.html
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /html/index.html HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.55.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/html
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: OPTIONS, HEAD, GET, PUT, POST, DELETE
< Access-Control-Allow-Headers: *
< Accept: */*
< Accept-Language: en-US, en
< Accept-Charset: ISO-8859-1, utf-8
< Connection: close
< Host: http://localhost:8080
< Content-Length: 226
<
﻿<html>
    <head>
        <title>Hello from HttpServerLite!</title>
    </head>
    <body>
        <div>
            <h3>Hello!</h3>
            <p>Hello from HttpServerLite!</p>
        </div>
    </body>
</html>* Closing connection 0
```

Code:
```
Console.WriteLine("Initializing server");
_Server = new Webserver(_Hostname, _Port, false, null, null, DefaultRoute);
_Server.Settings.Headers.Host = "http://" + _Hostname + ":" + _Port;
_Server.Events.ServerStarted += ServerStarted;
_Server.Events.ServerStopped += ServerStopped;
_Server.Events.ServerDisposing += ServerDisposing;
_Server.Events.Logger = Console.WriteLine;
_Server.Routes.Content.Add("./html/", true);
_Server.Routes.Content.Add("./img/", true);
_Server.Settings.Debug.Responses = true;
_Server.Settings.Debug.Routing = true;
_Server.Start();
```

**Comment by sqlnew** [comment]
Hi, I have written a simple test program to indicate that there is a bug. Please download this test program. I have uploaded it to the file saving website.
[https://fs.163.com/fs/display/?p=X-NETEASE-HUGE-ATTACHMENT&file=8T9cP5naUq1qY_8Yx_bxT7M9IqVIL2v5O_pRbTlG9q8aO14059R_cKssacjGkEmSx0ZuE9W-IyhuoqrHkjHb-Q&p=X-NETEASE-HUGE-ATTACHMENT&language=en](url)
![20211219013132](https://user-images.githubusercontent.com/5029826/146650568-883d6a0a-8957-45b1-99de-adc29e9e1c20.png)

**Comment by jchristn** [comment]
I don't see the same behavior as you.  When I build the HttpServerApp and run it, and then use cURL to GET the /getjson URL, I see:
```
C:\Users\joelc\Downloads\HttpServerApp\bin>curl -v http://localhost:7000/getjson
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 7000 (#0)
> GET /getjson HTTP/1.1
> Host: localhost:7000
> User-Agent: curl/7.55.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/plain;charset=utf-8
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: OPTIONS, HEAD, GET, PUT, POST, DELETE
< Access-Control-Allow-Headers: *
< Accept: */*
< Accept-Language: en-US, en
< Accept-Charset: ISO-8859-1, utf-8
< Connection: close
< Host: http://localhost:7000
< Content-Length: 65
<
{
  "biaoshi": "01010101345",
  "username": "123testtestabc"
}* Closing connection 0
```
Notice there is only one ```Content-Length``` header.

I also do not see the behavior you are experiencing when I use the Test project.

**Comment by sqlnew** [comment]
It's not the "/getjson" path that has a problem, but this one http://localhost:7000/html/hello.html There is a problem with the path. When reading and accessing local files, two content - length.

![20211220113002](https://user-images.githubusercontent.com/5029826/146707744-465e0743-c63a-42f0-8088-b77021bd832a.jpg)

**Comment by sqlnew** [comment]
After tracking the source code, it is found that the content length tag is added in two places:

![20211220151428](https://user-images.githubusercontent.com/5029826/146727320-2090351f-a562-4d72-8c51-d41248e8c67a.jpg)
![20211220151545](https://user-images.githubusercontent.com/5029826/146727339-059c59a9-5754-4189-aafc-bfb81d3d14b1.jpg)

![20211220151842](https://user-images.githubusercontent.com/5029826/146727547-0c42e7f3-08ce-4a12-b125-622c58fc6d59.jpg)

This "content length" flag has been repeatedly set in the function sendinternalasync

**Comment by jchristn** [comment]
Got it, thank you, was able to reproduce!  Fixing now.
```
C:\Users\joelc>curl -v http://localhost:7000/html/hello.html
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 7000 (#0)
> GET /html/hello.html HTTP/1.1
> Host: localhost:7000
> User-Agent: curl/7.55.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/html
< Content-Length: 262
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: OPTIONS, HEAD, GET, PUT, POST, DELETE
< Access-Control-Allow-Headers: *
< Accept: */*
< Accept-Language: en-US, en
< Accept-Charset: ISO-8859-1, utf-8
< Connection: close
< Host: http://localhost:7000
< Content-Length: 262
<
﻿<!DOCTYPE html>

<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>
    <meta charset="utf-8" />
    <title></title>
</head>
<body>
    <div>
        <span>Hello 你好！</span>
        <img src="open.jpg" />
    </div>
</body>
</html>* Closing connection 0
```

**Comment by jchristn** [comment]
Please confirm this is fixed in v1.2.1.4: https://www.nuget.org/packages/HttpServerLite/1.2.1.4
